### PR TITLE
[MIG][16.0] hr_timesheet_sheet

### PR DESCRIPTION
**Pull request by len-foss**
The migration needed a few fixes to tests and views, each in separate commits. 
Furthermore, create have been made to wok in batch and the deprecated dynamic onchange has been upgraded to a binary domain field.

### CODE REVIEW

File: hr_timesheet_sheet/views/hr_department_views.xml [hunk 1]
```diff
@@ -18,18 +18,18 @@
         <field name="name">hr.department.kanban.inherit</field>
         <field name="model">hr.department</field>
         <field name="inherit_id" ref="hr.hr_department_view_kanban" />
-        <field
-            name="groups_id"
-            eval="[(4,ref('hr_timesheet.group_hr_timesheet_user'))]"
-        />
```
Comment (AungKoKoLin1997) [inline hr_timesheet_sheet/views/hr_department_views.xml]: Why we need to remove this and add groups in each part?

Comment (pedrobaeza) [reply hr_timesheet_sheet/views/hr_department_views.xml]: This is something needed in v16. Please check the point "Any view with groups_id..." in https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-16.0

**Review by AungKoKoLin1997 — APPROVED** [review]
LGTM

**Comment by tarteo** [comment]
Have you considered this comment: https://github.com/OCA/timesheet/pull/580#issuecomment-1538653864

**Comment by astirpe** [comment]
Why opening this PR? There's already a migration in https://github.com/OCA/timesheet/pull/580

**Comment by len-foss** [comment]
@astirpe 
You're right! Simply because I missed it, it wasn't marked as a migration in the main issue.

**Pull request closed without merging**